collect_app/src/main/java/org/odk/collect/android/formentry/audit/AuditEventLogger.java

```diff
@@ -57,6 +58,8 @@ public void logEvent(AuditEvent.AuditEventType eventType, boolean writeImmediate
      */
     public void logEvent(AuditEvent.AuditEventType eventType, FormIndex formIndex,
                          boolean writeImmediatelyToDisk, String questionAnswer, long currentTime, String changeReason) {
+        checkAndroidUIThread();
```

Comment:
This should prevent us from making the same mistake again. It might feel a little risking as it could cause a production crash if we end up calling `logEvent` or `flush` from a background thread in a flow that our automated or regression tests don't cover. However, I think in that case, it'll be faster and easier to see what's happening than have to wait for a crash/bug that is created by the concurrent access.